--- gegl-0.4.2/gegl/module/geglmodule.c~	2018-05-13 11:48:27.000000000 +0000
+++ gegl-0.4.2/gegl/module/geglmodule.c	2018-05-20 21:32:31.992685959 +0000
@@ -354,7 +354,14 @@
 static gboolean
 gegl_module_open (GeglModule *module)
 {
-  module->module = g_module_open (module->filename, 0);
+  char avx2filename[PATH_MAX];
+  sprintf(avx2filename, 4095, "%s.avx2", module->filename);
+
+  if (__builtin_cpu_supports("avx2"))
+	module->module = g_module_open (avx2filename, 0);
+
+  if (! module->module)
+	  module->module = g_module_open (module->filename, 0);
 
   if (! module->module)
     {
